pandas/tools/tests/test_pivot.py

```diff
@@ -936,6 +936,35 @@ def test_crosstab_no_overlap(self):
 
         tm.assert_frame_equal(actual, expected)
 
+    def test_margin_ignore_dropna_bug(self):
+        # GH 12577
+        # pivot_table counts null into margin ('All')
+        # when margins=true and dropna=true
+
+        df = pd.DataFrame({'a': [1, 2, 2, 2, 2, np.nan],
+                           'b': [3, 3, 4, 4, 4, 4]})
+        actual = pd.crosstab(df.a, df.b, margins=True)
```

Comment:
explictly add ``dropna=True`` for these. tests

then add a test for ``dropna=False`` (same set of tests)
and you will find it raises. the ``level`` is ``None`` fails.

```
In [10]: pd.crosstab(df.a,df.b,margins=True,dropna=False)
KeyError: 'Level None not found'
```